pydantic/typing.py

```diff
@@ -288,6 +288,34 @@ def resolve_annotations(raw_annotations: Dict[str, Type[Any]], module_name: Opti
         except NameError:
             # this is ok, it can be fixed with update_forward_refs
             pass
+        except TypeError as te:
+            if value.__class__ is not ForwardRef:
+                raise
+
+            # In case of `ForwardRef`, we give it another shot if we want to leverage "new" annotations
+            # with older versions of python
+            try:
+                from future_typing import transform_annotation
+            except ImportError:
+                import warnings
+
+                warnings.warn(
+                    'If you try to use generic builtins or new `|` Union operator, '
+                    'you need to install `pydantic[future-annotations]` or directly `future-typing`.',
+                    UserWarning,
+                )
+                raise te
+            else:
+                # We need a merged version of typing (with typing_extensions) to
+                # ensure we can resolve things like `Literal` or `Annotated` with older versions
+                from . import typing_merge
```

Comment:
At first I did something like
```py
all_typing_mod_name = 'pydantic_all_typing___'
all_typing_mod = types.ModuleType(all_typing_mod_name)
exec("from typing import *\nfrom typing_extensions import *", all_typing_mod.__dict__)
```
But since using `eval()` is problematic with `cython`, I went with this approach.